src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -872,7 +882,7 @@ HeapWord* G1CollectedHeap::attempt_allocation_humongous(size_t word_size) {
     if (should_try_gc) {
       bool succeeded;
       result = do_collection_pause(word_size, gc_count_before, &succeeded,
-                                   GCCause::_g1_humongous_allocation);
+                                   GCCause::_g1_humongous_allocation, force_gc);
```

Comment:
I believe the reason for this GC if `force_gc` is true is *not* that we do not have enough space for the humongous allocation, but we are doing a "pre-emptive" GC because of the allocation of "word_size" similar to before.